src/Adapter/MSTest.CoreAdapter/Execution/TestExecutionManager.cs

```diff
@@ -247,11 +244,31 @@ private void ExecuteTestsInSource(IEnumerable<TestCase> tests, IRunContext runCo
                     sourceLevelParameters = sourceLevelParameters.Concat(this.sessionParameters).ToDictionary(x => x.Key, x => x.Value);
                 }
 
-                var sourceSettingsProvider = isolationHost.CreateInstanceForType(
-                    typeof(TestAssemblySettingsProvider),
-                    null) as TestAssemblySettingsProvider;
+                TestAssemblySettingsProvider sourceSettingsProvider = null;
+
+                try
+                {
+                    sourceSettingsProvider = isolationHost.CreateInstanceForType(
+                        typeof(TestAssemblySettingsProvider),
+                        null) as TestAssemblySettingsProvider;
+                }
+                catch (TypeLoadException ex)
+                {
+                    PlatformServiceProvider.Instance.AdapterTraceLogger.LogInfo(
+                        "Could not create TestAssemblySettingsProvider instance in child app-domain",
+                        ex);
+                }
+
+                TestAssemblySettings sourceSettings;
+                if (sourceSettingsProvider != null)
+                {
+                    sourceSettings = sourceSettingsProvider.GetSettings(source);
+                }
+                else
+                {
+                     sourceSettings = new TestAssemblySettings();
+                }
 
-                var sourceSettings = sourceSettingsProvider.GetSettings(source);
                 var parallelWorkers = sourceSettings.Workers;
```

Comment:
Ensure that default values for sourceSettings.Workers, & Scope is set appropriately in case we fail to create instance of "TestAssemblySettingsProvider"

Reply:
Yes, checked that.